### PR TITLE
chore(docs): Adding samples for GCE instance creation

### PR DESCRIPTION
**Pull request by m-strzelczyk**
Providing samples for the following docs page:
https://cloud.google.com/compute/docs/instances/create-start-instance

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Review by Xmasreturns — COMMENTED** [review]
Thanks for sending this over. I think most of my comments can apply across several methods, so the later versions of creating a VM can hopefully get updated with some find/replace.

File: samples/snippets/sample_create_vm.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+            This value uses following format: "projects/{project_name}/global/images/{image_name}"
+
+    Returns:
+        AttachedDisk object configured to be created as a copy of specified image.
+    """
+    boot_disk = compute_v1.AttachedDisk()
+    initialize_params = compute_v1.AttachedDiskInitializeParams()
+
+    initialize_params.source_image = source_image
+
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:64]: I was trying to make the "main" part of the method stand out. In this case it's setting the `source_image` value, below it's setting the `source_snapshot` value. All other params are basically the same, in both image and snapshot disk creation methods.

I don't have any strong feelings about this, so I can change if you think it's more confusing than helping.

File: samples/snippets/sample_create_vm.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+    initialize_params.disk_size_gb = disk_size_gb
+    initialize_params.disk_type = disk_type
+    boot_disk.initialize_params = initialize_params
+    boot_disk.auto_delete = True
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:68]: OK, I'll add comments to point out that it's important to configure it if you want to have the disk deleted with the VM.

File: samples/snippets/sample_create_vm.py [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+         disk_type: the type of disk you want to create. The value uses following format:
+            "zones/{zone}/diskTypes/(pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+            For example: "zones/us-west3-b/diskTypes/pd-ssd"
+        disk_size_gb: size of the new disk in gigabytes
+
+    Returns:
+        AttachedDisk object configured to be created as an empty disk.
+    """
+    disk = compute_v1.AttachedDisk()
+    init_params = compute_v1.AttachedDiskInitializeParams()
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:96]: Yes, you are right. It's better to be consistent.

File: samples/snippets/sample_create_vm.py [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+# [END compute_instances_create_from_image_plus_empty_disk]
+
+
+# [START compute_instances_create_from_image_plus_snapshot_disk]
+# [START compute_instances_create_from_snapshot]
+def disk_from_snapshot(
+    disk_type: str, disk_size_gb: int, boot: bool, snapshot_url: str
+) -> compute_v1.AttachedDisk():
+    """
+    Create an AttachedDisk object to be used in VM instance creation. Uses an Image as the
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:114]: You're right, it's using snapshot.

File: samples/snippets/sample_create_vm.py [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+# [START compute_instances_create_from_image_plus_empty_disk]
+# [START compute_instances_create_from_custom_image]
+# [START compute_instances_create_from_image]
+def create_with_disks(
+    project_id: str,
+    zone: str,
+    instance_name: str,
+    disks: List[compute_v1.AttachedDisk],
+    machine_type: str = "n1-standard-1",
+    # source_image: str = "projects/debian-cloud/global/images/family/debian-10",
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:158]: This should be removed :)

File: samples/snippets/sample_create_vm.py [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+    network_link: str = "global/networks/default",
+    subnetwork_link: str = None,
+) -> compute_v1.Instance:
+    """
+    Send an instance creation request to the Compute Engine API and wait for it to complete.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone you want to use. For example: “us-west3-b”
+        instance_name: name of the new virtual machine.
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:168]: OK. Yeah, I agree that it's messy.

File: samples/snippets/sample_create_vm.py [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+
+    # Collect information into the Instance object.
+    instance = compute_v1.Instance()
+    instance.name = instance_name
+    instance.disks = disks
+    full_machine_type_name = f"zones/{zone}/machineTypes/{machine_type}"
+    instance.machine_type = full_machine_type_name
+    instance.network_interfaces = [network_interface]
+
+    # Shielded Instance settings
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:199]: I admit I was a bit confused by the other samples (Cloud Console, gcloud and API) on the [docs page](https://cloud.google.com/compute/docs/instances/create-start-instance). They include the shielded settings sometimes and sometimes not. For example "Create a VM instance from a public image" section has `enableSecureBoot` config listed.

I personally don't care what values we put here, I just wanted to show that those values are there. I can just drop this set of configs and fall back to defaults. What do you think we should do with this? I feel like the Shielded VM parts were shoehorned there, and they don't really fit with the *basics* of instance creation.

File: samples/snippets/sample_create_vm.py [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+    return instance
+
+
+# [END compute_instances_create_from_custom_image]
+
+
+# [START compute_instances_create_from_image_plus_empty_disk]
+def create_with_additional_disk(project_id: str, zone: str, instance_name: str):
+    """
+    Create a new VM instance with Debian 10 operating system and a 11Gb additional
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply samples/snippets/sample_create_vm.py:291]: Yes, they use GB in Cloud Console, so I'm changing to that.

**Review by parthea — APPROVED** [review]

File: samples/snippets/sample_create_vm.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Xmasreturns) [reply samples/snippets/sample_create_vm.py:205]: Based on the text in the docs page for some examples, I see that Shielded VM info is denoted as "Optional:". There are also many notes on the page that describe particular behavior with regard to Shielded VMs. In that case, could we include the Shielded VM lines with default values, but comment them out? That way, they stay optional but visible.

**Pull request merged**